nightshift/comms/sierra_search_platform.py

```diff
@@ -0,0 +1,337 @@
+        elif self.json_response["data"]["suppressed"]:
+            return True
+        else:
+            return False
+
+
+class NypPlatform(PlatformSession):
+    def __init__(self) -> None:
+        """
+        Authenticates and opens a session with NYPL Platform.
+        Relies on credentials stored in evironmental variables.
+        """
+        client_id, client_secret, oauth_server, target = self._get_credentials()
+        token = self._get_token(client_id, client_secret, oauth_server)
+        agent = f"{__title__}/{__version__}"
+
+        PlatformSession.__init__(self, authorization=token, agent=agent, target=target)
+        logger.info("NYPL Platform session initiated.")
+
+    def _get_credentials(
+        self,
+    ) -> tuple[Optional[str], Optional[str], Optional[str], Optional[str]]:
+        """
+        Retrieves NYPL Platform credentials from environmental variables.
+
+        Returns:
+                (client_id, secret_id, oauth_server, platform_env)
+        """
+        return (
+            os.getenv("NYPL_PLATFORM_CLIENT"),
+            os.getenv("NYPL_PLATFORM_SECRET"),
+            os.getenv("NYPL_PLATFORM_OAUTH"),
+            os.getenv("NYPL_PLATFORM_ENV"),
+        )
+
+    def _get_token(
+        self,
+        client_id: Optional[str],
+        client_secret: Optional[str],
+        oauth_server: Optional[str],
+    ) -> PlatformToken:
+        """
+        Obtains an access token for NYPL Platform
+
+        Args:
+            client_id:                      NYPL Platfrom client id
+            client_secret:                  NYPL Platform client secret
+            oauth_server:                   NYPL Platform authorization server
+
+        Returns:
+            `PlatformToken` instance
+
+        Raises:
+            SierraSearchPlatformError
+        """
+        try:
+            token = PlatformToken(client_id, client_secret, oauth_server)
+            logger.info("NYPL Platform access token obtained.")
+            return token
+        except BookopsPlatformError as exc:
+            logger.error(f"Unable to obtain access token for NYPL Platform. {exc}")
+            raise SierraSearchPlatformError
+
+    def get_sierra_bib(self, sierraId: int) -> SearchResponse:
+        """
+        Searches NYPL Platform for a given sierra bib and returns its status
+
+        Args:
+            sierraId:                       Sierra bib number
+
+        Returns:
+            `SearchResponse` instance
+
+        Raises:
+            `ns_exceptions.SierraSearchPlatformError`
+        """
+        try:
+            response = self.get_bib(sierraId)
+            logger.debug(
+                f"NYPL Platform request ({response.status_code}): {response.url}."
+            )
+        except BookopsPlatformError as exc:
+            logger.error(
+                f"Error while querying NYPL Platform for Sierra bib # {sierraId}. {exc}"
+            )
+            raise SierraSearchPlatformError(exc)
+        else:
```

Comment:
just a nit, I prefer to omit this `else:` line, I think it's clearer that the lines below execute as part of the flow if they aren't in this block. YMMV though!

Reply:
Moved up lines under `else` to the block under `try`. Maybe there is a good reason to use `else` after `except` but I agree this reads better.